brainglobe_workflows/cellfinder/cellfinder_main.py

```diff
@@ -0,0 +1,305 @@
+
+
+def run_workflow_from_cellfinder_run(cfg):
+    """
+    Run workflow based on the cellfinder_core.main.main()
+    function.
+
+    The steps are:
+    1. Read the input signal and background data as two separate
+       Dask arrays.
+    2. Run the main cellfinder pipeline on the input Dask arrays,
+       with the parameters defined in the input configuration (cfg).
+    3. Save the detected cells as an xml file to the location specified in
+       the input configuration (cfg).
+
+    We plan to time each of the steps in the workflow individually,
+    as well as the full workflow.
+
+    Parameters
+    ----------
+    cfg : CellfinderConfig
+        a class with the required setup methods and parameters for
+        the cellfinder workflow
+    """
+    # Read input data as Dask arrays
+    signal_array = read_with_dask(cfg.signal_parent_dir)
+    background_array = read_with_dask(cfg.background_parent_dir)
+
+    # Run main analysis using `cellfinder_run`
+    detected_cells = cellfinder_run(
+        signal_array, background_array, cfg.voxel_sizes
+    )
+
+    # Save results to xml file
+    save_cells(detected_cells, cfg.detected_cells_filepath)
+
+
+def setup_workflow(default_config_dict: dict = default_config_dict):
+    """Prepare configuration to run workflow
+
+    This includes
+    - instantiating the config dictionary,
+    - checking if the input data exists locally, and fetching from
+      GIN repository otherwise,
+    - creating the directory for the output of the workflow if it doesn't exist
+
+    To instantiate the config dictionary, we first check if an environment
+    variable "CELLFINDER_CONFIG_PATH" pointing to a config json file exists.
+    If not, the default config is used.
+
+    Parameters
+    ----------
+    default_config_dict : dict
+        a dictionary with the default config parameters
+
+    Returns
+    -------
+    _type_
+        _description_
+    """
+
+    # Define config
+    if "CELLFINDER_CONFIG_PATH" in os.environ.keys():
+        input_config_path = Path(os.environ["CELLFINDER_CONFIG_PATH"])
+        assert input_config_path.exists()
+
+        # read into dict (assuming config is a json dict?)
+        # TODO:add error handling here?
+        with open(input_config_path) as cfg:
+            config_dict = json.load(cfg)
+
+        config = CellfinderConfig(**config_dict)
+
+        logging.info(
+            "Configuration retrieved from "
+            f'{os.environ["CELLFINDER_CONFIG_PATH"]}'
+        )
+
+    else:
+        config = CellfinderConfig(**default_config_dict)
+        logging.info("Using default configuration")
+
+    # Retrieve and add lists of input data to config if neither are defined
+    if not (config.list_signal_files and config.list_signal_files):
+        config = retrieve_input_data(config)
+
+    # Create output directory if it doesn't exist
+    # TODO: should I check if it exists and has data in it?
```

Comment:
maybe open an issue for discussion?

Reply:
potentially create a folder with the package name and timestamp, like `output_cellfinder_2023-10-04_09:17:56/` or is that overkill?

Reply:
will do, I think it's a good solution for now!